doc/source/whatsnew/v0.24.0.rst

```diff
@@ -436,6 +436,7 @@ Backwards incompatible API changes
 - Incorrectly passing a :class:`DatetimeIndex` to :meth:`MultiIndex.from_tuples`, rather than a sequence of tuples, now raises a ``TypeError`` rather than a ``ValueError`` (:issue:`24024`)
 - :func:`pd.offsets.generate_range` argument ``time_rule`` has been removed; use ``offset`` instead (:issue:`24157`)
 - In 0.23.x, pandas would raise a ``ValueError`` on a merge of a numeric column (e.g. ``int`` dtyped column) and an ``object`` dtyped column (:issue:`9780`). We have re-enabled the ability to merge ``object`` and other dtypes; pandas will still raise on a merge between a numeric and an ``object`` dtyped column that is composed only of strings (:issue:`21681`)
+- :class:`DatetimeIndex` and :class:`TimedeltaIndex` no longer ignore the dtype precision. Passing a non-nanosecond resolution dtype will raise a ``ValueError`` (:issue:`24753`)
```

Comment:
What do people think about this? In 0.23.4 we ignore the dtype precision

```python
In [3]: pd.DatetimeIndex(['2000'], dtype='datetime64[us]')
Out[3]: DatetimeIndex(['2000-01-01'], dtype='datetime64[ns]', freq=None)
```

Should that raise now? Or should we deprecate first?

Reply:
we actually can convert this (and I *think* we do?)

Reply:
though I would not be averse to deprecating for consistency reasons

Reply:
We do convert an *array* with non-ns precision dtype. That's similar to converting any sequence of datetime-like things

```python
In [7]: arr = np.array([1, 2], dtype='datetime64[us]')

In [8]: pd.DatetimeIndex(arr)
Out[8]: DatetimeIndex(['1970-01-01 00:00:00.000001', '1970-01-01 00:00:00.000002'], dtype='datetime64[ns]', freq=None)
```

The issue is when the user says `DatetimeIndex(..., dtype='datetime64[us]')`. That's like saying "I want the dtype of the *output* DatetimeIndex to be datetime64[us]", which isn't supported.